### ISSUE TITLE
[bug] AutotoolsToolchain doesn't pass -debug to msvc linker

### ISSUE THREAD

**Issue by dvirtz** [conan-io/conan#10338]
I was building a package with `-s build_type=Debug` and tried to debug the generated executable but the debugger complained that "Module was built without symbols". 
After a short investigation it seemed that `-debug` flag should have been added to `LDFLAGS`.
Adding the following to the recipe enabled me to debug the executable:

```
tc = AutotoolsToolchain(self)
if self._is_msvc and self.settings.build_type in ["Debug", "RelWithDebInfo"]:
    tc.ldflags.append("-debug")
tc.generate()
```

I tried to look at the code of the legacy `AutotoolsEnvironment` build helper and couldn't see it passing this flag either but I also couldn't find any issue reporting that so far so I'm not sure if it's broken as well.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: Visual Studio 16
  * Conan version: 1.44.0

### Logs (Executed commands with output) (Include/Attach if Applicable)

This is part of the generated conanautotoolstoolchain.bat file before the workaround
```
set CPPFLAGS=%CPPFLAGS%
set CXXFLAGS=%CXXFLAGS% -Zi -Ob0 -Od
set CFLAGS=%CFLAGS% -Zi -Ob0 -Od
set LDFLAGS=%LDFLAGS%
```

and after
```
set CPPFLAGS=%CPPFLAGS%
set CXXFLAGS=%CXXFLAGS% -Zi -Ob0 -Od
set CFLAGS=%CFLAGS% -Zi -Ob0 -Od
set LDFLAGS=%LDFLAGS% -debug
```

**Comment by memsharded** [comment]
Hi @dvirtz 

Thanks for reporting. I guess #10339 should fix it, but a quick question: Does this LDFLAG apply to all gcc-like compilers? The assumption there in code is ``if compiler in ["clang", "gcc", "apple-clang", "qcc", "mcst-lcc"]:``


Edit: At least clang doesn't seem to support ``-debug`` linker option: https://clang.llvm.org/docs/ClangCommandLineReference.html#linker-flags

UPDATE: sorry @dvirtz I went too fast for unit tests assuming a Nix platform, didn't read properly about Windows-msvc compiler. I'll double check now

**Comment by franramirez688** [comment]
Thanks for reporting the issue @dvirtz
This fix is going to be released at the next Conan 1.45 release.
